### ISSUE TITLE
VMM unable to create socket since 0.21

### ISSUE THREAD

**Issue by jeromegn** [firecracker-microvm/firecracker-go-sdk#217]
I keep getting the following messages when trying to start a VMM:

```
DEBU[0001] Called NewMachine()
DEBU[0001] Called Machine.Start()
DEBU[0001] Marking Machine as Started
DEBU[0001] Running handler validate.NetworkCfg
DEBU[0001] Running handler validate.JailerCfg
DEBU[0001] Running handler fcinit.SetupNetwork
DEBU[0001] Running handler fcinit.SetupKernelArgs
DEBU[0001] Running handler fcinit.StartVMM
INFO[0001] Called startVMM(), setting up a VMM on /home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root/run/firecracker.socket
DEBU[0001] VMM started socket path is /home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root/run/firecracker.socket

DEBU[0004] GET /machine-config HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/json
Accept-Encoding: gzip

<snip>

WARN[0004] Failed handler "fcinit.StartVMM": Firecracker did not create API socket /home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root/run/firecracker.socket: context deadline exceeded
```

I've upgraded my firecracker and jailer to 0.21.1, and upgraded this go package to 0.21.0. This worked fine with 0.20.

Here's a strace for the whole thing if that's of any interest:

<details><summary>strace</summary>

```
execve("/usr/local/bin/jailer-0.21.1", ["/usr/local/bin/jailer-0.21.1", "--id", "62fab4aa", "--uid", "123", "--gid", "100", "--exec-file", "/usr/local/bin/firecracker-0.21."..., "--node", "0", "--chroot-base-dir", "/home/jerome/src/github.com/supe"..., "--", "--seccomp-level", "0", "--api-sock", "/run/firecracker.socket"], 0x7fffc464e6e8 /* 57 vars */) = 0
mmap(NULL, 560, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fc873e38000
arch_prctl(ARCH_SET_FS, 0x7fc873e38140) = 0
set_tid_address(0x741d80)               = 1165392
rt_sigaction(SIGPIPE, {sa_handler=SIG_IGN, sa_mask=[], sa_flags=SA_RESTORER|SA_RESTART, sa_restorer=0x4cbf46}, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=0}, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [RT_1 RT_2], NULL, 8) = 0
rt_sigaction(SIGSEGV, {sa_handler=0x4b7440, sa_mask=[], sa_flags=SA_RESTORER|SA_ONSTACK|SA_SIGINFO, sa_restorer=0x4cbf46}, NULL, 8) = 0
rt_sigaction(SIGBUS, {sa_handler=0x4b7440, sa_mask=[], sa_flags=SA_RESTORER|SA_ONSTACK|SA_SIGINFO, sa_restorer=0x4cbf46}, NULL, 8) = 0
sigaltstack(NULL, {ss_sp=NULL, ss_flags=SS_DISABLE, ss_size=0}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fc873e36000
sigaltstack({ss_sp=0x7fc873e36000, ss_flags=0, ss_size=8192}, NULL) = 0
brk(NULL)                               = 0x1916000
brk(0x1917000)                          = 0x1917000
rt_sigprocmask(SIG_BLOCK, ~[RTMIN RT_1 RT_2], [], 8) = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
open("/proc/self/fd", O_RDONLY|O_CLOEXEC|O_DIRECTORY) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
getdents64(3, /* 6 entries */, 2048)    = 144
close(3)                                = 0
getdents64(3, 0x19161d8, 2048)          = -1 EBADF (Bad file descriptor)
close(3)                                = -1 EBADF (Bad file descriptor)
getrandom("\xb3\x96\x27\xb2\x61\x04\xb8\x9a\x4d\x88\x02\xf4\x24\x96\x58\x93", 16, GRND_NONBLOCK) = 16
brk(0x1918000)                          = 0x1918000
clock_gettime(CLOCK_PROCESS_CPUTIME_ID, {tv_sec=0, tv_nsec=456555}) = 0
open("/usr/local/bin/firecracker-0.21.1", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_PATH) = 3
readlink("/proc/self/fd/3", "/usr/local/bin/firecracker-0.21."..., 4095) = 33
fstat(3, {st_mode=S_IFREG|0755, st_size=2717512, ...}) = 0
stat("/usr/local/bin/firecracker-0.21.1", {st_mode=S_IFREG|0755, st_size=2717512, ...}) = 0
close(3)                                = 0
stat("/usr/local/bin/firecracker-0.21.1", {st_mode=S_IFREG|0755, st_size=2717512, ...}) = 0
open("/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_PATH) = 3
readlink("/proc/self/fd/3", "/home/jerome/src/github.com/supe"..., 4095) = 76
fstat(3, {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
stat("/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
close(3)                                = 0
stat("/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
mkdir("/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root", 0777) = -1 ENOENT (No such file or directory)
mkdir("/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa", 0777) = -1 ENOENT (No such file or directory)
mkdir("/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1", 0777) = 0
mkdir("/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa", 0777) = 0
mkdir("/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root", 0777) = 0
open("/usr/local/bin/firecracker-0.21.1", O_RDONLY|O_CLOEXEC) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
fcntl(3, F_GETFD)                       = 0x1 (flags FD_CLOEXEC)
fstat(3, {st_mode=S_IFREG|0755, st_size=2717512, ...}) = 0
open("/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root/firecracker-0.21.1", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0100755) = 4
fcntl(4, F_SETFD, FD_CLOEXEC)           = 0
fstat(4, {st_mode=S_IFREG|0755, st_size=0, ...}) = 0
fchmod(4, 0100755)                      = 0
copy_file_range(3, NULL, 4, NULL, 2717512, 0) = 2717512
close(4)                                = 0
close(3)                                = 0
open("/proc/mounts", O_RDONLY|O_CLOEXEC) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
brk(0x1919000)                          = 0x1919000
brk(0x191b000)                          = 0x191b000
brk(0x191c000)                          = 0x191c000
brk(0x191e000)                          = 0x191e000
brk(0x1924000)                          = 0x1924000
brk(0x1926000)                          = 0x1926000
read(3, "proc /proc proc rw,nosuid,nodev,"..., 8192) = 3103
read(3, "", 8192)                       = 0
close(3)                                = 0
mkdir("/sys/fs/cgroup/cpu,cpuacct/firecracker-0.21.1/62fab4aa", 0777) = 0
mkdir("/sys/fs/cgroup/cpuset/firecracker-0.21.1/62fab4aa", 0777) = 0
open("/sys/fs/cgroup/cpuset/firecracker-0.21.1/cpuset.cpus", O_RDONLY|O_CLOEXEC) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
fstat(3, {st_mode=S_IFREG|0644, st_size=0, ...}) = 0
read(3, "0-11\n", 32)                   = 5
read(3, "", 27)                         = 0
close(3)                                = 0
open("/sys/fs/cgroup/cpuset/firecracker-0.21.1/62fab4aa/cpuset.cpus", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
write(3, "0-11\n", 5)                   = 5
close(3)                                = 0
open("/sys/fs/cgroup/cpuset/firecracker-0.21.1/cpuset.mems", O_RDONLY|O_CLOEXEC) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
fstat(3, {st_mode=S_IFREG|0644, st_size=0, ...}) = 0
read(3, "0\n", 32)                      = 2
read(3, "", 30)                         = 0
close(3)                                = 0
open("/sys/fs/cgroup/cpuset/firecracker-0.21.1/62fab4aa/cpuset.mems", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
write(3, "0\n", 2)                      = 2
close(3)                                = 0
open("/sys/fs/cgroup/cpuset/firecracker-0.21.1/62fab4aa/cpuset.mems", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
write(3, "0\n", 2)                      = 2
close(3)                                = 0
open("/sys/devices/system/node/node0/cpulist", O_RDONLY|O_CLOEXEC) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
fstat(3, {st_mode=S_IFREG|0444, st_size=4096, ...}) = 0
read(3, "0-11\n", 4097)                 = 5
read(3, "", 4092)                       = 0
close(3)                                = 0
open("/sys/fs/cgroup/cpuset/firecracker-0.21.1/62fab4aa/cpuset.cpus", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
write(3, "0-11\n", 5)                   = 5
close(3)                                = 0
mkdir("/sys/fs/cgroup/pids/firecracker-0.21.1/62fab4aa", 0777) = 0
getpid()                                = 1165392
open("/sys/fs/cgroup/cpu,cpuacct/firecracker-0.21.1/62fab4aa/tasks", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
write(3, "1165392\n", 8)                = 8
close(3)                                = 0
open("/sys/fs/cgroup/cpuset/firecracker-0.21.1/62fab4aa/tasks", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
write(3, "1165392\n", 8)                = 8
close(3)                                = 0
open("/sys/fs/cgroup/pids/firecracker-0.21.1/62fab4aa/tasks", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
write(3, "1165392\n", 8)                = 8
close(3)                                = 0
unshare(CLONE_NEWNS)                    = 0
mount(NULL, "/", NULL, MS_REC|MS_SLAVE, NULL) = 0
mount("/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root", "/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root", NULL, MS_BIND|MS_REC, NULL) = 0
chdir("/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root") = 0
mkdir("old_root", 0600)                 = 0
pivot_root(".", "old_root")             = 0
chdir("/")                              = 0
umount2("old_root", MNT_DETACH)         = 0
rmdir("old_root")                       = 0
mkdir("/", 0777)                        = -1 EEXIST (File exists)
stat("/", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
chown("/", 123, 100)                    = 0
mkdir("/dev", 0777)                     = 0
chown("/dev", 123, 100)                 = 0
mkdir("/dev/net", 0777)                 = 0
chown("/dev/net", 123, 100)             = 0
mkdir("/run", 0777)                     = 0
chown("/run", 123, 100)                 = 0
mknod("/dev/net/tun", S_IFCHR|0600, makedev(0xa, 0xc8)) = 0
chown("/dev/net/tun", 123, 100)         = 0
mknod("/dev/kvm", S_IFCHR|0600, makedev(0xa, 0xe8)) = 0
chown("/dev/kvm", 123, 100)             = 0
rt_sigprocmask(SIG_BLOCK, ~[RTMIN RT_1 RT_2], [], 8) = 0
rt_sigprocmask(SIG_BLOCK, ~[], NULL, 8) = 0
setgid(100)                             = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
setgroups(0, NULL)                      = 0
rt_sigprocmask(SIG_BLOCK, ~[RTMIN RT_1 RT_2], [], 8) = 0
rt_sigprocmask(SIG_BLOCK, ~[], NULL, 8) = 0
setuid(123)                             = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigaction(SIGPIPE, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=SA_RESTORER|SA_RESTART, sa_restorer=0x4cbf46}, {sa_handler=SIG_IGN, sa_mask=[], sa_flags=SA_RESTORER|SA_RESTART, sa_restorer=0x4cbf46}, 8) = 0
execve("/firecracker-0.21.1", ["/firecracker-0.21.1", "--id", "62fab4aa", "--start-time-us", "352076567834", "--start-time-cpu-us", "456", "--seccomp-level", "0", "--api-sock", "/run/firecracker.socket"], 0x7ffcb4e12ef0 /* 57 vars */) = 0
mmap(NULL, 560, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fdec7c25000
arch_prctl(ARCH_SET_FS, 0x7fdec7c25140) = 0
set_tid_address(0x756540)               = 1165392
rt_sigaction(SIGPIPE, {sa_handler=SIG_IGN, sa_mask=[], sa_flags=SA_RESTORER|SA_RESTART, sa_restorer=0x510dbb}, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=0}, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [RT_1 RT_2], NULL, 8) = 0
rt_sigaction(SIGSEGV, {sa_handler=0x4a72e0, sa_mask=[], sa_flags=SA_RESTORER|SA_ONSTACK|SA_SIGINFO, sa_restorer=0x510dbb}, NULL, 8) = 0
rt_sigaction(SIGBUS, {sa_handler=0x4a72e0, sa_mask=[], sa_flags=SA_RESTORER|SA_ONSTACK|SA_SIGINFO, sa_restorer=0x510dbb}, NULL, 8) = 0
sigaltstack(NULL, {ss_sp=NULL, ss_flags=SS_DISABLE, ss_size=0}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fdec7c23000
sigaltstack({ss_sp=0x7fdec7c23000, ss_flags=0, ss_size=8192}, NULL) = 0
brk(NULL)                               = 0xfae000
brk(0xfaf000)                           = 0xfaf000
rt_sigprocmask(SIG_BLOCK, ~[RTMIN RT_1 RT_2], [], 8) = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigaction(SIGSYS, {sa_handler=0x4d8390, sa_mask=~[RTMIN RT_1 RT_2], sa_flags=SA_RESTORER|SA_SIGINFO, sa_restorer=0x510dbb}, NULL, 8) = 0
rt_sigaction(SIGBUS, {sa_handler=0x4d86f0, sa_mask=~[RTMIN RT_1 RT_2], sa_flags=SA_RESTORER|SA_SIGINFO, sa_restorer=0x510dbb}, NULL, 8) = 0
rt_sigaction(SIGSEGV, {sa_handler=0x4d86f0, sa_mask=~[RTMIN RT_1 RT_2], sa_flags=SA_RESTORER|SA_SIGINFO, sa_restorer=0x510dbb}, NULL, 8) = 0
brk(0xfb2000)                           = 0xfb2000
getrandom("\x71\x56\xec\xe9\x2b\xf4\xbf\x1b\x3e\xb7\x8f\x38\x49\xa5\x1b\x29", 16, GRND_NONBLOCK) = 16
eventfd2(0, EFD_NONBLOCK)               = 3
dup(3)                                  = 4
rt_sigprocmask(SIG_UNBLOCK, [RT_1 RT_2], NULL, 8) = 0
mmap(NULL, 2109440, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fdec7a20000
mprotect(0x7fdec7a22000, 2101248, PROT_READ|PROT_WRITE) = 0
rt_sigprocmask(SIG_BLOCK, ~[RTMIN RT_1 RT_2], [], 8) = 0
clone(child_stack=0x7fdec7c22998, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID|0x400000, parent_tid=[1165393], tls=0x7fdec7c22b20, child_tidptr=0x756540) = 1165393
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
epoll_create1(EPOLL_CLOEXEC)            = 6
epoll_ctl(6, EPOLL_CTL_ADD, 3, {EPOLLIN, {u32=1, u64=1}}) = 0
timerfd_create(CLOCK_MONOTONIC, TFD_CLOEXEC|TFD_NONBLOCK) = 8
epoll_ctl(6, EPOLL_CTL_ADD, 8, {EPOLLIN, {u32=2, u64=2}}) = 0
open("/dev/kvm", O_RDWR|O_CLOEXEC)      = 9
fcntl(9, F_SETFD, FD_CLOEXEC)           = 0
ioctl(9, KVM_GET_API_VERSION, 0)        = 12
ioctl(9, KVM_CHECK_EXTENSION, KVM_CAP_IRQCHIP) = 1
ioctl(9, KVM_CHECK_EXTENSION, KVM_CAP_IOEVENTFD) = 1
ioctl(9, KVM_CHECK_EXTENSION, KVM_CAP_IRQFD) = 1
ioctl(9, KVM_CHECK_EXTENSION, KVM_CAP_USER_MEMORY) = 1
ioctl(9, KVM_CHECK_EXTENSION, KVM_CAP_SET_TSS_ADDR) = 1
ioctl(9, KVM_CHECK_EXTENSION, KVM_CAP_NR_MEMSLOTS) = 509
ioctl(9, KVM_CREATE_VM, 0)              = 10
ioctl(9, KVM_GET_VCPU_MMAP_SIZE, 0)     = 12288
brk(0xfb3000)                           = 0xfb3000
ioctl(9, KVM_GET_SUPPORTED_CPUID, {nent=44, entries=[...]}) = 0
ioctl(9, KVM_GET_MSR_INDEX_LIST, 0xfb2840) = 0
eventfd2(0, EFD_NONBLOCK)               = 11
eventfd2(0, EFD_NONBLOCK)               = 12
eventfd2(0, EFD_NONBLOCK)               = 13
dup(11)                                 = 14
eventfd2(0, EFD_NONBLOCK)               = 15
dup(13)                                 = 16
dup(15)                                 = 17
epoll_pwait(6,
```

</details>

^ I had manually added `--api-sock /run/firecracker.socket` as a test here in my vendored package. With or without this, it doesn't work.

Directory structure:

```
$ tree -a -l -p -u -g /home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa
/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa
├── firecracker-0.21.1
│   └── 62fab4aa
│       └── root
└── [-rw-r--r-- 123      100 ]  rootfs.ext4

3 directories, 1 files

$ tree -a -l -p -u -g /home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root
/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root
├── dev
│   ├── [Dcrw------- 123      100 ]  kvm
│   └── net
│       └── [Dcrw------- 123      100 ]  tun
├── [-rwxr-xr-x root     0   ]  firecracker-0.21.1
└── run

3 directories, 3 files
```

The firecracker process is then left running, without crashing untill I kill it:

```
$ ps aux | grep firecracker
123      1165392  0.0  0.0   3620     4 pts/2    Sl   11:06   0:00 /firecracker-0.21.1 --id 62fab4aa --start-time-us 352076567834 --start-time-cpu-us 456 --seccomp-level 0 --api-sock /run/firecracker.socket
```

**Comment by kzys** [comment]
Thanks for reporting @jeromegn,

The socket path on the log (`/home/jerome/src/github.com/superfly/nomad-firecracker/firecrackers/62fab4aa/firecracker-0.21.1/62fab4aa/root/run/firecracker.socket` - 132 bytes) is pretty long and it would hit Linux's limit.

http://man7.org/linux/man-pages/man7/unix.7.html

> On Linux, sun_path is 108 bytes in size; see also NOTES, below.

We hit this limit before and added some workarounds in our tests. Can you try shorten the path?

https://github.com/firecracker-microvm/firecracker-go-sdk/blob/73e1707289db644a6ab59c7347322089c329b996/machine_test.go#L133

However, this doesn't explain why you didn't hit the issue pre-0.21 though.

**Comment by jeromegn** [comment]
Shortening the path fixes it.

I remember this was not a problem a while ago. I just realized this had been a problem since a few versions using this specific code path, but that it wasn't in a different one because it resulted in shorter socket paths.

Would it be possible to validate the length of the socket path? This should be an error or else it's very confusing.

It's also a very long path by default. It adds multiple subdirectories. I wish I could specific the exact location where I want the jail to live. I'm running this with nomad and if I want to keep things nice and tidy, I need to set the jailer path within the "allocation directory". This directory is garbage collected by nomad automatically. However, its base length is quite long (`/var/lib/nomad/alloc/7c6fc6e6-7d48-5642-6b0b-5c1d657b75f8/app/...`). I think I can shorten the jail path enough that it will fit in 108 bytes though.

**Comment by jeromegn** [comment]
SocketPath is also hardcoded now when using the jailer: https://github.com/firecracker-microvm/firecracker-go-sdk/blob/master/jailer.go#L291

If I could use something like `api.sock` instead of `/run/firecracker.socket`, I would save a few bytes there too! Unfortunately whatever I set in my `MachineConfig` is thrown away.

**Comment by kzys** [comment]
Firecracker is able to show the error like below;

```
% firecracker --api-sock /tmp/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.sock
2020-03-17T16:40:25.927549966 [anonymous-instance:ERROR:src/firecracker/src/main.rs:66] Firecracker panicked at 'Error creating the HTTP server: IOError(Custom { kind: InvalidInput, error: "path must be shorter than SUN_LEN" })', src/libcore/result.rs:1165:5
2020-03-17T16:40:25.943202426 [anonymous-instance:ERROR:src/firecracker/src/main.rs:76]    0: firecracker::main::{{closure}}
   1: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:477
   2: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:380
   3: rust_begin_unwind
             at src/libstd/panicking.rs:307
   4: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
   5: core::result::unwrap_failed
             at src/libcore/result.rs:1165
   6: api_server::ApiServer::bind_and_run
   7: std::sys_common::backtrace::__rust_begin_short_backtrace
   8: core::ops::function::FnOnce::call_once{{vtable.shim}}
   9: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/liballoc/boxed.rs:922
  10: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/liballoc/boxed.rs:922
      std::sys_common::thread::start_thread
             at src/libstd/sys_common/thread.rs:13
      std::sys::unix::thread::Thread::new::thread_start
             at src/libstd/sys/unix/thread.rs:79

```

Instead of doing the same check on the SDK side, I personally prefer to either show the message as is, or modify Firecracker to provide a mechanism which is easier to consume from the SDK (e.g. different exit code, format logs in JSON, ...)

**Comment by kzys** [comment]
While #219 hasn't been merged, #230 has fixed the issue.
